simple_use_case/cwl/make_gmsh_mesh.cwl

```diff
@@ -3,24 +3,30 @@
 cwlVersion: v1.0
 class: CommandLineTool
 doc: |
-  mesh generation with gmsh
-hints:
-  DockerRequirement:
-    dockerPull: wftools-simpleusecase
+  Generate the computational mesh with gmsh
 baseCommand: ["gmsh"]
-arguments: ["-$(inputs.dimension)", "$(inputs.geofile.path)", "-o", "$(inputs.outfilename)"]
+arguments: ["-2", "$(inputs.geofile.path)", "-o", "mesh.msh"]
+stdout: output.txt
+requirements:
+  InlineJavascriptRequirement: {}
 inputs:
   geofile:
     type: File
     doc: "Geometry file in a gmsh-readable format"
-  outfilename:
-    type: string
-    doc: "The name of the mesh file to be written, including the desired extension"
-  dimension:
-    type: int
-    doc: "The desired dimension of the mesh"
 outputs:
   mesh:
     type: File
     outputBinding:
-      glob: $(inputs.outfilename)
+      glob: "mesh.msh"
+  domain_size:
+    type: float
+    outputBinding:
+      glob: "output.txt"
+      loadContents: true
+      outputEval: |
+        ${
+            var output = self[0].contents;
+            var size_string = output.split("Used domain size:")[1];
+            size_string = size_string.split("Used mesh size")[0];
+            return parseFloat(size_string);
+        }
```

Comment:
So basically the workflow tool needs to be able to capture `stdout` of the gmsh command which just prints this information. I think this should be possible with the other tools as well and I will have a look at how to do this with pydoit, snakemake, nextflow and AiiDA